client/lib/purchases/assembler.js

```diff
@@ -82,26 +72,22 @@ function createPurchaseObject( purchase ) {
 	};
 
 	if ( 'credit_card' === purchase.payment_type ) {
-		const payment = Object.assign( {}, object.payment, {
+		const payment = {
+			...object.payment,
 			creditCard: {
 				id: Number( purchase.payment_card_id ),
 				type: purchase.payment_card_type,
 				processor: purchase.payment_card_processor,
 				number: purchase.payment_details,
 				expiryDate: purchase.payment_expiry,
-				expiryMoment: purchase.payment_expiry
-					? i18n.moment( purchase.payment_expiry, 'MM/YY' )
-					: null,
 			},
-		} );
+		};
 
-		return Object.assign( {}, object, { payment } );
+		return { ...object, payment };
```

Comment:
Here, to avoid recreating the large `object`, I would simply assign to `object.payment.creditCard`:
```js
if ( isCreditCard ) {
  object.payment.creditCard = ...;
}
```
That should work equally well.

Reply:
Good idea, thank you!